src/main/java/Duke.java

```diff
@@ -1,10 +1,213 @@
+                    "  " + taskDLine + "\nNow you have " +
+                    taskList.getSize() + " tasks in the list." );
+            storage.writeListIntoFile(taskList.getList());
+            break;
+        case DELETE:
+            try {
+                String strDelete = sc.nextLine();
+                if (strDelete.isEmpty()) {
+                    throw new DukeDescriptionException("Empty Description");
+                }
+                int delete = Integer.parseInt(strDelete.substring(1)); //there must be a space between command and input
+                if (delete <= 0 || delete > taskList.getSize()) {
+                    throw new BadDescriptionException("Description for delete cannot be " + delete);
+                }
+                Task deletedTask = taskList.getTask(delete - 1);
+                taskList.removeTask(delete - 1);
+                calender.removeTask(deletedTask, deletedTask.getDate());
+                System.out.println("Okay! I have deleted this task:\n" +
+                        "  " + deletedTask + "\nNow you have " +
+                        taskList.getSize() + " tasks in the list.");
+                storage.writeListIntoFile(taskList.getList());
+            } catch (NumberFormatException e) {
+                throw new BadDescriptionException("Non-Integer");
+            }
+            break;
+        case SEARCH:
+            String search = sc.nextLine();
+            if (search.isEmpty()) {
+                throw new DukeDescriptionException("Empty Description");
+            }
+            LocalDate date = Parser.dateParser(search.substring(1));
+            System.out.println("Here are the events on " +
+                    date.format(Parser.DATE_READ_FORMATTER) + ":");
+            calender.searchDate(date);
+            break;
+        case FIND:
+            String find = sc.nextLine();
+            if (find.isEmpty()) {
+                throw new DukeDescriptionException("Empty Description");
+            }
+            TaskList outputList = new TaskList(taskList.find(find.substring(1)));
+            System.out.println("Here are the matching tasks in your list:");
+            outputList.printTask();
+            break;
+        default:
+            break;
+        }
+    }
+
+    /**
+     * Generates DukeProject. This method will run the main interface of the DukeProject.
+     * It will read commands given by the system using a scanner and store results in a task list before writing it
+     * into storage.
+     */
+    public void run() {
+        ui.getIntroduction();
+        while (sc.hasNext()) {
+            String next = sc.next();
+            if (next.equals("bye")) {
+                break;
+            }
+            else {
+                try {
+                    ui.lineBreak();
+                    Command command = Parser.commandParser(next);
+                    readCommand(command);
+                } catch (InvalidCommandException e) {
```

Comment:
Polymorphism can be applied here. Simply catch the different types of Exceptions with the | operator and then print out the Exception message accordingly via .getMessage(), which can also help in code maintenance for longer term projects.